packages/neon-core/src/sc/ContractParam.ts

```diff
@@ -435,18 +501,31 @@ export class ContractParam implements NeonObject<ContractParamLike> {
     if (this.type === parseEnum(other.type, ContractParamType)) {
       switch (this.type) {
         case ContractParamType.Array:
-        case ContractParamType.Map:
           if (
             Array.isArray(this.value) &&
             Array.isArray(other.value) &&
             this.value.length === other.value.length
           ) {
-            return this.value.every((cp, i) =>
+            return (this.value as ContractParam[]).every((cp, i) =>
               cp.equals((other.value as ContractParamLike[])[i])
             );
           }
           return false;
 
+        case ContractParamType.Map:
+          if (
+            Array.isArray(this.value) &&
+            Array.isArray(other.value) &&
+            this.value.length === other.value.length
+          ) {
+            return (this.value as ContractParamMap).every(

```

Comment:
this assumes that the map is ordered. Is this the case? (u have no tests for this at the moment)

Reply:
a `map` on the VM side is ordered [ref](https://github.com/neo-project/neo-vm/blob/5b0a39811b34abacab1273f3ee5a9a9f7e52ac7b/src/Neo.VM/Types/Map.cs#L21)